tests/v2/validation/provisioning/hosted/aks/provisioning/hosted_provisioning_test.go

```diff
@@ -0,0 +1,116 @@
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type HostedClusterProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	standardUserClient *rancher.Client
+}
+
+func (h *HostedClusterProvisioningTestSuite) TearDownSuite() {
+	h.session.Cleanup()
+}
+
+func (h *HostedClusterProvisioningTestSuite) SetupSuite() {
+	testSession := session.NewSession(h.T())
+	h.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(h.T(), err)
+
+	h.client = client
+
+	enabled := true
+	var testuser = provisioning.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(h.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(h.T(), err)
+
+	h.standardUserClient = standardUserClient
+}
+
+func (h *HostedClusterProvisioningTestSuite) TestProvisioningHostedAKSCluster() {
+	tests := []struct {
+		name   string
+		client *rancher.Client
+	}{
+		{"Admin User", h.client},
+		{"Standard User", h.standardUserClient},
+	}
+
+	for _, tt := range tests {
+		h.Run(tt.name, func() {
+			subSession := h.session.NewSession()
+			defer subSession.Cleanup()
+
+			client, err := tt.client.WithSession(subSession)
+			require.NoError(h.T(), err)
+
+			cloudCredential, err := azure.CreateAzureCloudCredentials(client)
+			require.NoError(h.T(), err)
+
+			clusterName := provisioning.AppendRandomString("akshostcluster")
+			clusterResp, err := aks.CreateAKSHostedCluster(client, clusterName, cloudCredential.ID, false, false, false, false, map[string]string{})
+			require.NoError(h.T(), err)
+
+			opts := metav1.ListOptions{
+				FieldSelector:  "metadata.name=" + clusterResp.ID,
+				TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+			}
+			watchInterface, err := h.client.GetManagementWatchInterface(management.ClusterType, opts)
+			require.NoError(h.T(), err)
+
+			checkFunc := clusters.IsHostedProvisioningClusterReady
+
+			err = wait.WatchWait(watchInterface, checkFunc)
+			require.NoError(h.T(), err)
+			assert.Equal(h.T(), clusterName, clusterResp.Name)
+
+			clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
+			require.NoError(h.T(), err)
+			assert.NotEmpty(h.T(), clusterToken)
+
+			updatedCluster, err := ScalingAKSNodePools(client, clusterResp, clusterName, cloudCredential.ID)
```

Comment:
@caliskanugur @markusewalker  so there will be some copy code, but should the scaling be it's own test case? Ie `(h func (h *HostedClusterProvisioningTestSuite) TestProvisioningHostedAKSClusterNodeScaling()` in case we want to add other validations to the main provisioning test. I would love to hear your opinions.

Reply:
I don't mind doing that, makes sense if we elect to add other validations to the main provisioning test. I'm open to whatever the group thinks is best.

Reply:
I think that makes sense to me. Code duplication can be prevented by doing this in a private method for the test suite as a cluster provisioning part. 

@markusewalker if you check the upgrade, you can do something like testSingleUpdate, or RBAC tests are also a good example of that. 

This private method I mentioned can be something like that. One cluster provisioning private method of the test suite used in:

- the provisioning test alone
- the scale bits followed after the provisioning test

Reply:
That's a great idea. It's your call Markus, I would be ok with the duplicate for now, with the scaling bit, and come back when you add more test cases, or do it now. Your call. But I do think it should be it's own test case.

Reply:
@igomez06 @caliskanugur pushed changes that refactors based on Ugur's suggestion. Also, I went and did the same for EKS and GKE as well.